src/Debugger/Impl/rtvs/R/packages.R

```diff
@@ -1,27 +1,86 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See LICENSE in the project root for license information.
 
-matrix.as.lists <- function(matrix.data) {
-    res <- list()
-    dn <- dimnames(matrix.data)
-    for (r in dn[[1]]) {
-        res[[r]] <- as.list(matrix.data[r,])
-    }
-    res
+packages.installed <- function() {
+  pkgs <- installed.packages(fields = c('Title', 'Author', 'Description'))
+  pkgs <- pkgs[!duplicated(pkgs[, 'Package']),]
+  apply(pkgs, 1, as.list)
 }
 
-packages.installed <- function() {
-    matrix.as.lists(installed.packages(fields = c('Title', 'Author', 'Description')))
+packages.loaded <- function() {
+  as.list(.packages())
 }
 
-packages.available <- function() {
-    matrix.as.lists(available.packages())
+packages.libpaths <- function() {
+  as.list(.libPaths())
 }
 
-packages.loaded <- function() {
-    as.list(.packages())
+download.packages.rds <- function(repo.url) {
+  # download the repository's packages.rds and return path to the temp file
+  pkg.rds.url <- gsub("src/contrib", "web/packages/packages.rds", repo.url)
+  pkg.rds.path <- tempfile('pkg', fileext = '.rds')
+  suppressWarnings(download.file(pkg.rds.url, destfile = pkg.rds.path, quiet = TRUE, mode = "wb"))
```

Comment:
I mean, if we are offline and then go online, can we refresh automatically or can user somehow do it?